agent/proxy.py

```diff
@@ -233,14 +237,26 @@ def remove_redirect(self, host):
             # default domain
             os.rmdir(host_directory)
 
+    def is_nginx_reloading(self) -> bool:
+        shutting_down_processes = int(
+            self.execute(
+                "ps ef | grep 'nginx: worker process is shutting down' | grep -v grep | wc -l"
+            )
+        )
+        if shutting_down_processes > 0:
+            return True
+
     @step("Reload NGINX")
-    def reload_nginx(self):
+    def reload_nginx(self, skip_if_reloading=False):
+        if skip_if_reloading and self.is_nginx_reloading():
+            return
         return self.execute("sudo systemctl reload nginx")
 
     @step("Generate NGINX Configuration")
     def generate_proxy_config(self):
         return self._generate_proxy_config()
 
+    @cached(cache=TTLCache(maxsize=128, ttl=10))
     def _generate_proxy_config(self):
```

Comment:
Should this really be cached? Rendering template doesn't cost much and you might end up not re-generating when it's actually required.

Reply:
ah yeah, good catch. Left this in after testing. Not needed.